### ISSUE TITLE
Fabric: Incompatible with Carpet Mod

### ISSUE THREAD

**Issue by tesseract-two** [Tuinity/Moonrise#24]
Moonrise crashes with Carpet Mod.

Environment:
````
Minecraft 1.21
Fabric Loader 0.16.5
Moonrise 0.1.0-beta.2+44f8058
Carpet 1.4.147+v240613
No Telemetry 1.8.0
````
Carpet's `Explosion_scarpetEventMixin` appears to conflict with Moonrise's `collisions.ExplosionMixin`. Setting `"required": false` in Carpet's `carpet.mixins.json` allow the game to launch and run fine, and explosions appear to work normally.

**Comment by Astropulse** [comment]
Additionally Architectury API has a similar issue with `architectury.mixins.json:MixinExplosion` conflicting with `collisions.ExplosionMixin`

**Comment by jpenilla** [comment]
Is this reproducible in 1.21.3?

**Comment by Assaro** [comment]
Yes it also crashes in 1.21.3
[crash-2024-11-23_22.56.01-client.txt](https://github.com/user-attachments/files/17882684/crash-2024-11-23_22.56.01-client.txt)

**Comment by jpenilla** [comment]
Can you try with disabling optimizedTNT in the carpet config? https://github.com/gnembon/fabric-carpet/blob/27ccd69c86c25357a2385c47db731b43ec5338b6/src/main/java/carpet/CarpetSettings.java#L231

**Comment by Assaro** [comment]
> Can you try with disabling optimizedTNT in the carpet config? https://github.com/gnembon/fabric-carpet/blob/27ccd69c86c25357a2385c47db731b43ec5338b6/src/main/java/carpet/CarpetSettings.java#L231

Does this need a recompiled jar?

**Comment by hitech95** [comment]
> Can you try with disabling optimizedTNT in the carpet config? https://github.com/gnembon/fabric-carpet/blob/27ccd69c86c25357a2385c47db731b43ec5338b6/src/main/java/carpet/CarpetSettings.java#L231

Yes even disabling the optionin carpet the mixin is still failing to apply.
